sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -284,7 +284,7 @@ private void exportCurrentBatch() {
 
       try {
         final CompletableResultCode result = spanExporter.export(batch);
-        result.join(exporterTimeoutMillis, TimeUnit.MILLISECONDS);
+        result.join(exporterTimeoutNanos, TimeUnit.NANOSECONDS);
```

Comment:
Nanos feels a bit like overkill here. There are no processors on which you can actually join at nanosecond precision are there? Or even less than microsecond?

Reply:
Timeouts are always handled with nanos against `System.nanoTime()` (wall time is what doesn't have less-than-microsecond precision). So what we pass gets converted into a nanosecond value. I agree it's overkill to ever use this, but the advantage of using nanos here is no need to document anything like "values are internally converted to millis precision", and I guess there isn't really a disadvantage.